### PR TITLE
engine(dm): use same enum value as dmpb for engine task stage

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #7115

### What is changed and how it works?
- now we uses the same grafana dashboard for dm in OP and engine(https://github.com/pingcap/tiflow/pull/7881), so we need to use same enum value as dmpb for engine task stage, to avoid confusion

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by maxshuang — APPROVED** [review]

**Comment by D3Hunter** [comment]
this changed the value of `TaskStage` which will be marshall into string, so there should not be compatibile issues

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: engine/jobmaster/dm/metadata/job.go [hunk 1]
```diff
@@ -28,19 +28,22 @@ import (
...
-	StageRunning
-	StagePaused
-	StageFinished
-	StageError
-	StagePausing
-	// UnScheduled means the task is not scheduled.
+	StageInit     TaskStage = iota + 1  // = 1 = dmpb.Stage_New
+	StageRunning                        // = 2 = dmpb.Stage_Running
+	StagePaused                         // = 3 ~= dmpb.Stage_Paused. in engine this stage means paused by user, if it's auto-paused by error, it's StageError
+	StageFinished TaskStage = iota + 2  // = 5 = dmpb.Stage_Finished. skip 4 - Stopped, no such stage in engine, see dm/worker/metrics.go
```
[the remark this answers is not in the document]
Comment (D3Hunter) [reply engine/jobmaster/dm/metadata/job.go:43]: yes, but below ones will not assign values automatically.

**Pull request merged**